core/state/intra_block_state.go

```diff
@@ -771,20 +771,33 @@ func (sdb *IntraBlockState) CreateAccount(addr common.Address, contractCreation
 		}
 
 		err = sdb.tracer.CaptureAccountWrite(addr)
-		if sdb.trace {
+		if sdb.trace && err != nil {
 			log.Error("error while CaptureAccountWrite", "err", err)
 		}
 	}
 
 	var previous *stateObject
+	var prevInc uint64
 	if contractCreation {
 		previous = sdb.getStateObject(addr)
+		if previous != nil {
+			prevInc = previous.data.Incarnation
```

Comment:
I don't think we want this. By our convention, the incarnation of any non-contract account is 0. So in this case prevInc would always be 0, which can be incorrect in the case where a contract has self-destructed, then Eth sent to it, then it got recreated again. We want to call `stateReader.ReadAccountIncarnation` in any case

Reply:
Right, I'll call stateReader.ReadAccountIncarnation in any case